marlowe-cli/cli-test/Language/Marlowe/CLI/Test/CLI/Types.hs

```diff
@@ -97,120 +101,133 @@ anyCLIMarloweThreadInputsApplied
-        [ "permanent" .= permanent
-        , "publisher" .= publisher
-        ]
+      [ "referenceCurrent"
+          .= Aeson.object
+            [ "permanent" .= permanent
+            , "publisher" .= publisher
+            ]
       ]
 
 -- | On-chain test operations for the Marlowe contract and payout validators.
-data CLIOperation =
-    -- | We use "private" currency minting policy which
+data CLIOperation
+  = -- | We use "private" currency minting policy which
     -- | checks for a signature of a particular issuer.
     Initialize
-    {
-      coMinLovelace         :: Lovelace                -- ^ Minimum lovelace to be sent to the contract.
-    , coContractNickname    :: Maybe ContractNickname  -- ^ The name of the wallet's owner.
-    , coRoleCurrency        :: Maybe CurrencyNickname  -- ^ If contract uses roles then currency is required.
-    , coContractSource      :: Contract.Source         -- ^ The Marlowe contract to be created.
-    , coSubmitter           :: Maybe WalletNickname    -- ^ A wallet which gonna submit the initial transaction.
-    , coMarloweValidators   :: Maybe MarloweValidators
-    }
+      { coMinLovelace :: Lovelace
+      -- ^ Minimum lovelace to be sent to the contract.
+      , coContractNickname :: Maybe ContractNickname
+      -- ^ The name of the wallet's owner.
+      , coRoleCurrency :: Maybe CurrencyNickname
+      -- ^ If contract uses roles then currency is required.
+      , coContractSource :: Contract.Source
+      -- ^ The Marlowe contract to be created.
+      , coSubmitter :: Maybe WalletNickname
+      -- ^ A wallet which gonna submit the initial transaction.
+      , coMarloweValidators :: Maybe MarloweValidators
+      }
   | Prepare
-    {
-      coContractNickname     :: Maybe ContractNickname      -- ^ The name of the contract.
-    , coInputs               :: [ParametrizedMarloweJSON]   -- ^ Inputs to the contract.
-    , coMinimumTime          :: SomeTimeout
-    , coMaximumTime          :: SomeTimeout
-    , coOverrideMarloweState :: Maybe M.State               -- ^ Useful for testing scenarios with non standard initial state.
-    }
-  | Publish                                          -- ^ Publishing can be a part of `Initialize` operation but we can also test it separately.
-    { coPublisher          :: Maybe WalletNickname   -- ^ Wallet used to cover fees. Falls back to faucet wallet.
-    , coPublishPermanently :: Maybe Bool             -- ^ Whether to publish script permanently.
-    }
+      { coContractNickname :: Maybe ContractNickname
+      -- ^ The name of the contract.
+      , coInputs :: [ParametrizedMarloweJSON]
+      -- ^ Inputs to the contract.
+      , coMinimumTime :: SomeTimeout
+      , coMaximumTime :: SomeTimeout
+      , coOverrideMarloweState :: Maybe M.State
+      -- ^ Useful for testing scenarios with non standard initial state.
+      }
+  | -- | Publishing can be a part of `Initialize` operation but we can also test it separately.
```

Comment:
Interesting that `fourmolu` knows how to correct haddock comments in sum types.

Reply:
It parses the AST and re-renders it (then parses again to check the AST hasn't changed)